File: 백준/Silver/10845. 큐/큐.py
from collections import deque

n = int(input())

cmd = []
q = deque()

for _ in range(n):
    cmd.append(input())

for i in cmd:
    if i == "pop":
        if len(q) == 0:
            print(-1)
        else:
            k = q.popleft()
            print(k)
    elif i == "size":
        print(len(q))
    elif i == "empty":
        if len(q) == 0:
            print(1)
        else:
            print(0)
    elif i == "front":
        if len(q) == 0:
            print(-1)
        else:
            print(q[0])
    elif i == "back":
        if len(q) == 0:
            print(-1)
        else:
            print(q[-1])
    else:
        p, m = i.split()
        q.append(m)

File: 백준/Silver/10845. 큐/큐.java
import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) throws Exception {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		int n = Integer.parseInt(br.readLine());
		LinkedList<Integer> queue = new LinkedList<>();
		
		for (int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String cmd = st.nextToken();
			
			switch(cmd) {
			case "push":
				queue.add(Integer.parseInt(st.nextToken()));
				break;
			case "pop":
				sb.append(queue.isEmpty() ? -1 : queue.poll()).append("\n");
				break;
			case "size":
				sb.append(queue.size()).append("\n");
				break;
			case "empty":
				sb.append(queue.isEmpty() ? 1 : 0).append("\n");
				break;
			case "front":
				sb.append(queue.isEmpty() ? -1 : queue.peekFirst()).append("\n");
				break;
			case "back":
				sb.append(queue.isEmpty() ? -1 : queue.peekLast()).append("\n");
				break;
			}
		}
		
		System.out.print(sb);
	}

}
